spec/unit/resource/remote_file_spec.rb

```diff
@@ -125,6 +125,31 @@
       expect { resource.source([]) }.to raise_error(ArgumentError)
     end
 
+    describe "when download the file from a source and server was down" do
+      node = Chef::Node.new
+      events = Chef::EventDispatch::Dispatcher.new
+      let(:path) { File.join(Dir.tmpdir, make_tmpname("chef_remote_file")) }
+      let(:run_context) { Chef::RunContext.new(node, {}, events) }
+      let(:retriable_resource) do
+        retriable_resource = Chef::Resource::RemoteFile.new(path, run_context)
+        retriable_resource
+      end
+
+      before do
+        retriable_resource.source("https://FQDN_THAT_IS_DOWN/file")
```

Comment:
this should probably not be a URL that fails on DNS lookups but something like `https://127.0.0.1:0/` which should always immediately RST very fast.

Reply:
I have removed this spec, because now it won't raise this error from `remote_file` resource. As this is handled in `lib/chef/http.rb`